web/api/servers.go

```diff
@@ -978,6 +989,215 @@ func editServerDataAdmin(c *gin.Context) {
+	ns := &services.Node{DB: db}
+	bs := &services.Backup{DB: db}
+	name := c.Query("name")
+	node := &server.Node
+
+	resolvedPath := "/daemon/server/" + strings.TrimPrefix(c.Request.URL.Path, "/api/servers/")
+	if c.Request.URL.RawQuery != "" {
+		resolvedPath += "?" + c.Request.URL.RawQuery
+	}
+
+	callResponse, err := ns.CallNode(node, c.Request.Method, resolvedPath, c.Request.Body, c.Request.Header)
+	if callResponse.StatusCode == http.StatusBadRequest { //If its a local node, the err will not be set, have to check the status code
+		newHeaders := cleanHttpReturnErrors(callResponse.Header)
+
+		c.DataFromReader(callResponse.StatusCode, callResponse.ContentLength, callResponse.Header.Get("Content-Type"), callResponse.Body, newHeaders)
+		c.Abort()
+		return
+	}
+	if response.HandleError(c, err, http.StatusInternalServerError) {
+		return
+	}
+
+	responseData := &pufferpanel.ServerBackupResponse{}
+	err = json.NewDecoder(callResponse.Body).Decode(responseData)
+	if response.HandleError(c, err, http.StatusBadRequest) {
+		return
+	}
+
+	backup := &models.Backup{Name: name, FileName: responseData.BackupFileName, FileSize: responseData.FileSize, ServerID: server.Identifier}
+	err = bs.Create(backup)
+	if response.HandleError(c, err, http.StatusBadRequest) {
+		return
+	}
+
+	c.Status(http.StatusOK)
+}
+
+// @Summary Delete backup
+// @Description Removes the backup and its assosicated file
+// @Success 204 {object} nil
+// @Param id path string true "Server ID"
+// @Param backupId path string true "Backup ID"
+// @Router /api/servers/{id}/backup/Delete/{backupId} [delete]
+// @Security OAuth2Application[server.backup.delete]
+func deleteBackup(c *gin.Context) {
+	server := getServerFromGin(c)
+	db := middleware.GetDatabase(c)
+	ns := &services.Node{DB: db}
+	bs := &services.Backup{DB: db}
+	node := &server.Node
+
+	var err error
+	var backupId uint
+	if backupId, err = cast.ToUintE(c.Param("backupId")); err != nil {
+		response.HandleError(c, err, http.StatusBadRequest)
+		return
+	}
+
+	backup, err := bs.GetById(backupId)
+	if response.HandleError(c, err, http.StatusInternalServerError) {
+		return
+	}
+	if backup == nil {
+		c.Status(http.StatusNotFound)
+		return
+	}
+
+	resolvedPath := "/daemon/server/" + server.Identifier + "/backup" + "?fileName=" + backup.FileName
+
+	callResponse, err := ns.CallNode(node, "DELETE", resolvedPath, nil, nil)
+	if callResponse.StatusCode == http.StatusBadRequest { //If its a local node, the err will not be set, have to check the status code
+		newHeaders := cleanHttpReturnErrors(callResponse.Header)
+
+		c.DataFromReader(callResponse.StatusCode, callResponse.ContentLength, callResponse.Header.Get("Content-Type"), callResponse.Body, newHeaders)
+		c.Abort()
+		return
+	}
+	if response.HandleError(c, err, http.StatusInternalServerError) {
+		return
+	}
+
+	err = bs.Delete(backupId)
+	if response.HandleError(c, err, http.StatusBadRequest) {
+		return
+	}
+
+	c.Status(http.StatusOK)
```

Comment:
If there is no body, use StatusNoContent instead

Reply:
fixed in [3afa64c](https://github.com/pufferpanel/pufferpanel/pull/1379/commits/3afa64cabbf2a66b294418fec744d28d5024bb52)